modules/custom/activity_creator/src/ActivityNotifications.php

```diff
@@ -140,36 +213,63 @@ public function changeStatusOfActivity(Activity $activity, $status = ACTIVITY_ST
    * @param \Drupal\Core\Session\AccountInterface $account
    *   Account object.
    * @param array $status
-   *   Array of statuses.
-   * @param \Drupal\Core\Entity\EntityBase $entity
-   *   Optionally provide a related entity to get the activities for.
+   *   Array of notification statuses.
    *
    * @return array
-   *   Returns an array of notification ids.
+   *   Returns an array of notification ids or empty array.
    */
-  private function getNotificationIds(AccountInterface $account, array $status = [], EntityBase $entity = NULL) {
-    $destinations = ['notifications'];
-
-    $uid = $account->id();
-
-    $entity_query = \Drupal::entityQuery('activity');
-    $entity_query->condition('field_activity_recipient_user', $uid, '=');
-    $entity_query->condition('field_activity_destinations', $destinations, 'IN');
-
-    if ($entity !== NULL) {
-      $entity_type = $entity->getEntityTypeId();
-      $entity_id = $entity->id();
-      $entity_query->condition('field_activity_entity.target_id', $entity_id, '=');
-      $entity_query->condition('field_activity_entity.target_type', $entity_type, '=');
+  protected function getNotificationIds(AccountInterface $account, array $status = []): array {
+    // Get the user ID.
+    if (!empty($uid = $account->id())) {
+      try {
+        $query = $this->database->select('activity_notification_status', 'ans')
+          ->fields('ans', ['aid'])
+          ->condition('uid', $uid);
 
+        if (!empty($status)) {
+          $query->condition('status', $status, 'IN');
+        }
+        return $query->execute()->fetchCol();
+      }
+      catch (\Exception $exception) {
+        // Log the exception to watchdog.
+        $this->loggerFactory->get('default')->error($exception->getMessage());
+      }
     }
-    if (!empty($status)) {
-      $entity_query->condition('field_activity_status', $status, 'IN');
+    else {
+      return [];
     }
+  }
+
+  /**
+   * Deletes all entries in activity_notification_table by given ids.
+   *
+   * @param array $activity_ids
+   *   Array of activity ids to be deleted.
+   *
+   * @return bool
+   *   Status of update query.
+   */
+  public function deleteNotificationsbyIds(array $activity_ids): bool {
+    if (!empty($activity_ids)) {
+      // The transaction opens here.
+      $txn = $this->database->startTransaction();
+      try {
+        $this->database->delete('activity_notification_status')
+          ->condition('aid', $activity_ids)
+          ->execute();
+      }
+      catch (\Exception $exception) {
+        // Something went wrong somewhere, so roll back now.
+        $txn->rollBack();
+        // Log the exception to watchdog.
+        $this->loggerFactory->get('default')->error($exception->getMessage());
```

Comment:
@navneet0693, all the Behat tests are failing. Looks like it's because of this line.

https://travis-ci.org/goalgorilla/open_social/builds/600691656

Reply:
@bramtenhove I am on it.